azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventHubPartitionPump.java

```diff
@@ -211,7 +211,6 @@ public void onReceive(Iterable<EventData> events)
 		@Override
 		public void onError(Throwable error)
```

Comment:
Client retry probably re-established the connection in your experiment and that is how it recovered. Try disabling the retry, that would stop the pump.